### ISSUE TITLE
Metric: Available Space

### ISSUE THREAD

**Issue by AnalogJ** [AnalogJ/scrutiny#35]

**Comment by bergernetch** [comment]
This could be complicated for drives used with lvm and such

**Comment by AnalogJ** [comment]
Yeah, and I had to do a pretty deep dive into `ioctl` because I wanted to support accessing drives by their device file, rather than depending on their mount point (which would make the containerized collector more complicated). 

The eventual Window support will add another layer of complication as well. 

This is definitely not going to be available with the v1.0.0 release, but maybe someday.

**Comment by AnalogJ** [comment]
- https://stackoverflow.com/questions/20108520/get-amount-of-free-disk-space-using-go
- https://github.com/shirou/gopsutil/
- [stackoverflow.com/a/48490543](https://stackoverflow.com/a/48490543)
- ioctl
- https://github.com/golang/go/issues/33966
- [dswarbrick/smart/search?unscoped_q=ioctl (ioctl)](https://github.com/dswarbrick/smart/search?q=ioctl&unscoped_q=ioctl)
- [https://github.com/mdlayher/block/blob/master/block_linux.go](https://github.com/mdlayher/block/blob/master/block_linux.go?rgh-link-date=2020-09-18T15%3A13%3A52Z)

**Comment by Tone866** [comment]
would love to see that feature!
